libraries/appframeworks/portals/src/lib/store/index.ts

```diff
@@ -0,0 +1,126 @@
+import { configureStore } from "@reduxjs/toolkit";
+import type { EnhancedStore } from "@reduxjs/toolkit";
+import mappingSlice from "./slices/mapping";
+import layersSlice from "./slices/layers";
+import uiSlice from "./slices/ui";
+import {defaultLayerMap } from "../utils/defaultLayerMap";
+import measurementsSlice from "./slices/measurements";
+import { createLogger } from "redux-logger";
+import { persistReducer } from "redux-persist";
+import localForage from "localforage";
+import type { LayerMap } from "../types";
+
+console.log("store initializing ....");
+
+const customAppKey = new URLSearchParams(window.location.hash).get("appKey");
+
+const devToolsEnabled =
+  new URLSearchParams(window.location.search).get("devToolsEnabled") === "true";
+console.log("devToolsEnabled:", devToolsEnabled);
+const stateLoggingEnabledFromSearch = new URLSearchParams(
+  window.location.search,
+).get("stateLoggingEnabled");
+
+const inProduction = process.env.NODE_ENV === "production";
+
+console.log("in Production Mode:", inProduction);
+const stateLoggingEnabled =
+  (stateLoggingEnabledFromSearch !== null &&
+    stateLoggingEnabledFromSearch !== "false") ||
+  !inProduction;
+
+console.log(
+  "stateLoggingEnabled:",
+  stateLoggingEnabledFromSearch,
+  "x",
+  stateLoggingEnabled,
+);
+const logger = createLogger({
+  collapsed: true,
+});
+
+let middleware;
+if (stateLoggingEnabled === true) {
+  middleware = (getDefaultMiddleware) =>
+    getDefaultMiddleware({
+      serializableCheck: false,
+    }).concat(logger);
+} else {
+  middleware = (getDefaultMiddleware) =>
+    getDefaultMiddleware({
+      serializableCheck: false,
+    });
+}
+
+interface PortalStoreConfig {
+  APP_KEY?: string;
+  STORAGE_PREFIX?: string;
+  layerMap: LayerMap;
+}
+
+export const configurePortalStore = ({
```

Comment:
Store handling should, after some extensive exploration to make this work, probably be part of App functionality, not part of lib.